src/Raven.Server/Documents/DocumentsStorage.cs

```diff
@@ -782,10 +782,7 @@ public void DeleteTombstonesBefore(string collection, long etag, Transaction tra
                 return;
             }
 
-            foreach (var result in table.SeekBackwardFrom(_tombstonesSchema.FixedSizeIndexes["CollectionEtags"], etag))
-            {
-                table.Delete(result.Id);
```

Comment:
This is an easy mistake to make. Let us add code that will throw an error in this case.
In we continue iterating after an explicit delete, like working with collections